File: Dataset/Leetcode/train/15/40.java
 class Solution {
    public List<List<Integer>> XXX(int[] nums) {
        if (nums.length < 3) {
            return new ArrayList();
        }
        List<List<Integer>> result = new ArrayList<>();
        Arrays.sort(nums);
        int flag1 = -1;
        labela: while (++flag1<=nums.length-3) {
            int flag2=flag1+1;
            int flag3=nums.length - 1;
            while (flag2 < flag3) {
                if (nums[flag1] + nums[flag2] + nums[flag3] == 0) {
                    List<Integer> list = new ArrayList<>();
                    list.add(nums[flag1]);
                    list.add(nums[flag2]);
                    list.add(nums[flag3]);
                    String key =String.valueOf(nums[flag1])+String.valueOf(nums[flag2])+String.valueOf(nums[flag3]);
                    if(!mapStr.containsKey(key)){
                        result.add(list);
                        mapStr.put(key,"");
                    }

                    continue labela;
                }

                if (nums[flag1] + nums[flag2] + nums[flag3] > 0) {
                    flag3--;
                }

                if (nums[flag1] + nums[flag2] + nums[flag3] < 0) {
                    flag2++;
                }
            }

        }
        return result;
    }
    private static Map<String,String> mapStr =new HashMap<>();
}

File: Dataset/Leetcode/train/67/40.cpp
 class Solution {
public:
    string XXX(string a, string b) {
        int alen=a.size(),blen=b.size(),c=0;
        string result="";
        while(alen&&blen){
            if(a[alen-1]=='1' && b[blen-1]=='1' && !c){  //1 1 0
                result+="0";
                c=1;
            }
            else if(a[alen-1]=='1' && b[blen-1]=='1' && c){  //1 1 1
                result+="1";
                c=1;
            }
            else if((a[alen-1]-'0' || b[blen-1]-'0') && !c){  //1 0 0
                result+="1";
                c=0;
            }
            else if((a[alen-1]-'0' || b[blen-1]-'0') && c){  //1 0 1
                result+="0";
                c=1;
            }
            else if(a[alen-1]=='0' && b[blen-1]=='0' && !c){  //0 0 0
                result+="0";
                c=0;
            }
            else if(a[alen-1]=='0' && b[blen-1]=='0' && c){  //0 0 1
                result+="1";
                c=0;
            }
            alen--;
            blen--;
        }
        while(alen){
            if(a[alen-1]=='1' && c){  //1 1 
                result+="0";
                c=1;
            }
            else if(a[alen-1]-'0' || c){  //0 1    1 0
                result+="1";
                c=0;
            }
            else if(a[alen-1]=='0' && !c){  //0 0
                result+="0";
                c=0;
            }
            alen--;
        }
        while(blen){
            if(b[blen-1]=='1' && c){  //1 1 
                result+="0";
                c=1;
            }
            else if(b[blen-1]-'0' || c){  //0 1    1 0
                result+="1";
                c=0;
            }
            else if(b[blen-1]=='0' && !c){  //0 0
                result+="0";
                c=0;
            }
            blen--;
        }
        if(c)
            result+="1";
        reverse(result.begin(),result.end());
        return result;
    }
};

